examples/demo33/src/integration-test/groovy/demo/JsonControllerSpec.groovy

```diff
@@ -1,18 +1,29 @@
 package demo
 
-import geb.spock.GebSpec
 import grails.testing.mixin.integration.Integration
+import spock.lang.Specification
+
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.time.Duration
 
 @Integration
-class JsonControllerSpec extends GebSpec {
+class JsonControllerSpec extends Specification {
 
     void "test a json view is rendered"() {
+        given:
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create("http://localhost:$serverPort/json/index"))
+            .timeout(Duration.ofMinutes(2))
+            .header("Accept", "application/json")
+            .GET()
+            .build()
+
         when:
-        go '/json/index'
+        def response = HttpClient.newBuilder().build().send(request, HttpResponse.BodyHandlers.ofString())
 
         then:
-        downloadBytes() { HttpURLConnection connection ->
-            connection.setRequestProperty('Accept', 'application/json')
-        } == '{"foo":"bar"}'.bytes
+        response.body() == '{"foo":"bar"}'
     }
```

Comment:
Actually, we can even do it even simpler (because we are groovy people):
```groovy
    void 'test a json view is rendered'() {
        given:
        URL url = new URL("http://localhost:$serverPort/json/index")

        expect:
        url.text == '{"foo":"bar"}'
    }
```

Reply:
This won't add the accept header though?  I originally tried with URL, but then realized it requires that header.

Reply:
Well, that controller isn't going to return anything but json anyway. Have you tried it?

Reply:
If json is the only view, it works.  If you add a an html view, it fails.  I just expanded the test coverage since I think the intent of this test was to ensure the json view was returned when the Accept header was json